### ISSUE TITLE
Mission reset to WP1 on failsafe with failsafe_mission = OFF

### ISSUE THREAD

**Issue by Airwide** [iNavFlight/inav#5954]
## Current Behavior
While flying a waypoint mission with failsafe_mission = OFF, the mission gets reset to WP1 after the lost rc link has been reestablished:

<img width="1225" alt="image" src="https://user-images.githubusercontent.com/3070959/87790476-e560a600-c840-11ea-9d58-7501056fab50.png">

## Steps to Reproduce

1. Set failsafe_mission = OFF
2. Engage Waypoint mission.
3. Reduce rc link range by standing between the aircraft and the transmitter.
4. Turn around once the rc link has been lost to reestablish the link.

## Expected behavior
The mission is expected to continue even if rc link has been lost.

## Suggested solution(s)
I haven't been able to find the bug, but I see two possible ways to resolve the issue:
1. Prevent the temporary flight mode change during rc link reestablishment.
2. Store the current waypoint in a variable which is called during waypoint mission initiation.

## Additional context
- Aircraft: Sonicmodell AR Wing
- TX: FrSky Taranis X9D+
- RX: FrSky X4R-SB
- FC: Mateksys F405WING
- CLI dump: https://pastebin.com/gWD4R75n
- Blackbox log file to big to be attached.

### INAV version
- INAV/MATEKF405SE 2.5.1 Jun 16 2020 / 12:50:26 (098a574b3)
- GCC-9.2.1 20191025 (release) [ARM/arm-9-branch revision 277599]

**Comment by Airwide** [comment]
I reproduced the issue this morning. However, on the first of two failsafes the mission was not reset. Instead it was the ARM flight mode flag that was temporarily when the rc link was reestablished:

<img width="1060" alt="image" src="https://user-images.githubusercontent.com/3070959/87933303-4d0b3100-ca8d-11ea-948d-debbbe250d12.png">

On the second failsafe, the mission was reset to WP1 just before I reached wp2 as can be seen in below screenshot.
<img width="1060" alt="image" src="https://user-images.githubusercontent.com/3070959/87933488-9fe4e880-ca8d-11ea-80dd-f6ea48a5a0c8.png">

Link to log file: https://www.dropbox.com/s/yao96h9drdnvze8/LOG00001.TXT?dl=0
